### DISCUSSION TITLE
How to place key switches on a modified version of the PCB?

### DISCUSSION THREAD

**Discussion by valeriosetti** [strayer/taira-keyboard#7] [General]
Hi!
I've been using a "standard" ErgoDash for a couple of months now and, albeit I feel quite OK with it, I would like to make some changes. The 1st one would be for sure using the low profile keys instead of the taller/standard ones (that's how I come across this project, btw :) ).
I liked your idea to use PCB plates also as a replacement for the acrylic case, since I've done some research online, but I didn't find any low-cost option to cut plastic-acrylic plates, so perhaps PCB is the best option.

Now it comes my question: how did you manage to get all the keys properly aligned on the circuit's PCB as well as the top plate?
I mean, the only option I know when working with KiCAD is to import some vector graphics as background and then draw PCB's edges and place components directly on KiCAD. However I didn't see any SVG file on your repo... Am I missing something here?

Thanks in advance

**Comment by strayer** [comment]
Hey, sorry for taking so long to respond. Started a new job this year and was quite busy with that.

> I liked your idea to use PCB plates also as a replacement for the acrylic case, since I've done some research online, but I didn't find any low-cost option to cut plastic-acrylic plates, so perhaps PCB is the best option.

They are also just much *much* thinner than acrylic plates. At least all the options I could find where at least 3mm compared to the usual PCB thickness of 1.6mm.

But keep in mind that especially the top plate can be expensive to order. See here for example where JLCPCB charged 13.50 € extra because of the complexity of the top plate:

<img width="808" alt="grafik" src="https://user-images.githubusercontent.com/310624/212543496-8fd56224-57a3-4550-b5ee-c3a2545c8ed9.png">

This is because the mill has to run *a lot* of steps to cut out each individual key socket. Obviously not calculated into the standard JLCPCB pricing because this is not common for PCBs. Bottom plate and PCB where ~13 €. And this is all not yet including shipping and taxes.

I still greatly prefer the PCB plates to acrylic plates.

> Now it comes my question: how did you manage to get all the keys properly aligned on the circuit's PCB as well as the top plate?
I mean, the only option I know when working with KiCAD is to import some vector graphics as background and then draw PCB's edges and place components directly on KiCAD. However I didn't see any SVG file on your repo... Am I missing something here?

It has been a while, but I *think* this is how I did it:

First of all: I reused the switch footprint from the Corne keyboard when designing the Taira. The Corne also has a footprint called `SW_Hole` that is just the mounting hole for the top plate. What I did was after finishing the PCB I simply copied the whole thing into the top plate project, removed all unnecessary stuff (everything except the switches and mounting holes, which still remain in the schematic) and then changed the switch footprint from the actual switches to the SW_Hole. Then you can just let KiCad rebuild the PCB… the holes will be at the correct position and everything else will be removed. Then I modified the PCB outline a bit because the top plate is not covering the microcontroller areas.

The bottom plate was even simpler because it is exactly the same as the PCB, except there are NO components except the mounting holes.

There may have been some manual cleanup work, but I distinctly remember being surprised how easy it was to design the plates.

> Thanks in advance

Thanks for checking out the Taira! I assume this won't really help you for building replacement plates for an ErgoDash since the whole setup is a bit different. I guess you could theoretically do this too by doing the same to the ErgoDash PCB project and designing a custom switch hole footprint. I found the footprint editor to be surprisingly simple, even though I'm really just beginner-level with KiCad myself.

**Reply by valeriosetti** [reply]
> Hey, sorry for taking so long to respond. Started a new job this year and was quite busy with that.

No problem at all! Good for you BTW and thanks for the reply :)

> They are also just much much thinner than acrylic plates

Another intersting point that I missed when I first looked at this option.

> But keep in mind that especially the top plate can be expensive to order

According to the prices you listed this would still be cheaper than the acrylic, at least for the prices I found.

> What I did was after finishing the PCB I simply copied the whole thing into the top plate project, removed all unnecessary stuff (everything except the switches and mounting holes, which still remain in the schematic) and then changed the switch footprint from the actual switches to the SW_Hole.

So basically you did the switch placement for the PCB (I mean, the one with traces and components) manually, right?
Because my main problem was how to do the initial placement for the switches. Once that is done, then I would agree with your solution to design the top plate,

BTW recently I came across this nice project, [Ergogen](https://github.com/ergogen/ergogen), which looks quite powerful for building everything in a programmatical way.  Maybe I'll give it a try and see if that somehow fits better than manual placement :)
